internal/status-updater/controllers/node/controller.go

```diff
@@ -95,10 +98,47 @@ func (c *NodeController) pruneTopologyNodes() error {
 		validNodeTopologyCMMap[topology.GetNodeTopologyCMName(node.Name)] = true
 	}
 
+	var multiErr error
 	for _, cm := range nodeTopologyCms.Items {
-		if _, ok := validNodeTopologyCMMap[cm.Name]; !ok {
-			util.LogErrorIfExist(c.kubeClient.CoreV1().ConfigMaps(viper.GetString(constants.EnvTopologyCmNamespace)).Delete(context.TODO(), cm.Name, metav1.DeleteOptions{}), fmt.Sprintf("Failed to delete node topology cm %s", cm.Name))
+		_, ok := validNodeTopologyCMMap[cm.Name]
+		multiErr = multierror.Append(multiErr, c.pruneTopologyConfigMap(&cm, ok))
+	}
+
+	return nil
+}
+
+func (c *NodeController) pruneTopologyConfigMap(cm *v1.ConfigMap, isValidNodeTopologyCM bool) error {
+	if !isValidNodeTopologyCM {
+		util.LogErrorIfExist(c.kubeClient.CoreV1().ConfigMaps(viper.GetString(constants.EnvTopologyCmNamespace)).Delete(context.TODO(), cm.Name, metav1.DeleteOptions{}), fmt.Sprintf("Failed to delete node topology cm %s", cm.Name))
+	}
+
+	nodeTopology, err := topology.FromNodeTopologyCM(cm)
+	if err != nil {
+		return fmt.Errorf("failed to parse node topology cm %s: %v", cm.Name, err)
+	}
+
+	for i := range nodeTopology.Gpus {
+		nodeTopology.Gpus[i].Status.PodGpuUsageStatus = topology.PodGpuUsageStatusMap{}
+
+		// Remove non-existing pods from the allocation info
+		allocatingPodExists, err := isPodExist(c.kubeClient, nodeTopology.Gpus[i].Status.AllocatedBy.Pod, nodeTopology.Gpus[i].Status.AllocatedBy.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to check if pod %s exists: %v", nodeTopology.Gpus[i].Status.AllocatedBy.Pod, err)
 		}
+
+		if !allocatingPodExists {
+			nodeTopology.Gpus[i].Status.AllocatedBy = topology.ContainerDetails{}
+		}
+	}
```

Comment:
just a thought - in scale scenarios you could have 10000+ "gpus"
maybe we want to support it in a pool of goroutines, say 1 for each 100/1000 etc.
i'm not sure how much this section takes but it looked like it might be costly on scale.

if not - ignore.
if yes and you want to do in another jira - that's also ok open the jira and send me

Reply:
This section happens only upon initialization and is not part of the ongoing logic of the status-updater so I think we can start with that and improve if needed later on